printf: handle Array and Map

flavio opened the pull request:
Handle printing of `%v` with Array and Map objects.

Obviously this won't cover 100% of the cases, but it's better than nothing

flavio commented:
@fiji-flo hi, any chance to get this PR reviewed? :pleading_face:

fiji-flo commented on src/printf.rs:
```diff
@@ -307,6 +310,36 @@ mod test {
+        // a straight comparison
+        assert!(s.contains("hello:world"));
+        assert!(s.contains("number:42"));
+        assert!(Regex::new(r"^foo \{.*\}$").unwrap().is_match(&s));
```
We could check `starts_with("foo {")` and `ends_with("}")` to skip the regex dependency.

fiji-flo reviewed (changes requested):
Thanks for this PR @flavio 

Could you make it behave like golang `fmt.Printf` (see https://play.golang.org/p/3AR2Vlf7FWc) and remove the regex dependency?

I'll get back to you faster next time. I had some busy weeks.

flavio commented:
@fiji-flo, sorry for the late response. I was on vacation :)

I've updated the PR to address your notes. Note well, the rendering of arrays was already fine (unless I missed something), I fixed the rendering of maps.

Once you're happy with the code I can squash the 3 commits into a single one, unless you prefer to merge the branch as it is

fiji-flo commented on src/printf.rs:
```diff
@@ -307,6 +309,44 @@ mod test {
+        let s = s.unwrap();
+        // The print order is unpredictable, we can't write
+        // a straight comparison
+        assert!(s.contains("hello:world"));
+        assert!(s.contains("number:42"));
+        assert!(s.starts_with("foo map["));
+        assert!(s.ends_with("]"));
```
```suggestion
    assert!(s == "foo map[number:42 hello:world]" || s == "foo map[hello:world number:42]");
```
That's simpler and tests the white space between pairs. (Potentially needs formatting)

fiji-flo reviewed (changes requested):
Looks great, thanks a lot. If you could adopt the nit pick there I'll merge and create a 0.7.1 release 🥂

flavio commented:
@fiji-flo I've accepted the change, fixed the indentation, ran the tests and then squashed everything into a single commit

Thanks for your reviews!

fiji-flo reviewed (approved):
👍 Thanks. I'll fix the clippy warnings (that's on me) and get it released.

flavio commented:
@fiji-flo just out of curiosity, how do you handle release? I wanted to subscribe to release events to get notified about the `0.7.1` release, but I noticed no release has ever done and no git tags are associated to the repo :thinking:

fiji-flo commented:
I've never automated it. That's on my TODO for all my crates to move this into a github action. But I published it already.

I'll move to branches for the `0.X` and tags for releases then.
